src/components/colors-legend.components.tsx

```diff
@@ -19,11 +21,16 @@ import { useSelectedRoutes } from '../selected-routes.context'
 const ColorsLegend: FC<MapControlProps> = ({ position = 'topright' }) => {
   const { colors } = useRouteColors()
   const { routes, setRoutes } = useSelectedRoutes()
+  const [isLegendCollapsed, setLegendCollapseState] = useState<boolean>(false)
 
   if (!allRoutes || Object.keys(colors).length === 0) {
     return null
   }
 
+  const collapseLegend = () => {
+    setLegendCollapseState(!isLegendCollapsed)
+  }
```

Comment:
This function implies inversion of previous state so `toggleLegendCollapseState` is a better name.
```suggestion
  const toggleLegendCollapseState = () => {
    setLegendCollapseState(s => !s)
  }
```